lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -259,6 +260,37 @@ defmodule ExUnit.Runner do
     end
   end
 
+  defp maybe_create_tmp_dir(test, context, %{tmp_dir: true}) do
+    create_tmp_dir!(test, "", context)
+  end
+
+  defp maybe_create_tmp_dir(test, context, %{tmp_dir: path}) when is_binary(path) do
+    create_tmp_dir!(test, path, context)
+  end
+
+  defp maybe_create_tmp_dir(_, _, %{tmp_dir: other}) do
+    raise ArgumentError, "expected :tmp_dir to be atom true or a string, got: #{inspect(other)}"
+  end
+
+  defp maybe_create_tmp_dir(_, context, _) do
+    context
+  end
+
+  defp create_tmp_dir!(test, extra_path, context) do
+    module = escape_path(inspect(test.module))
+    name = escape_path(to_string(test.name))
+    path = Path.join(["tmp", module, name, extra_path])
```

Comment:
I think it could be handy to be able to configure the base directory. For example: `config :ex_unit, root_tmp_dir: System.get_env("TMPDIR", "mytmp")`. Thoughts?

Reply:
I thought the same but I don't have a use case for it yet, so I am glad to sit on it until a need arrives.

Reply:
This might be one: if you run your tests inside a Docker container and you mount the root directory, having the option to write to a directory that is not synced with the local machine might give a little boost in performance 🙂

Reply:
Ah, I see! If someone wants to explore this, it would be welcome. In particular, if we can have some numbers that show it is indeed better, it will help us move it forward. :+1: